target_chains/sui/scripts/pyth/create_price_feed.ts

```diff
@@ -1,99 +1,53 @@
-/// We build a programmable txn to create a price feed.
-import dotenv from "dotenv";
-
-import {
-  RawSigner,
-  SUI_CLOCK_OBJECT_ID,
-  TransactionBlock,
-  JsonRpcProvider,
-  Ed25519Keypair,
-  Connection,
-} from "@mysten/sui.js";
-
-dotenv.config({ path: "~/.env" });
-
-import { REGISTRY, NETWORK } from "../registry";
-
-// ================== Network dependent settings ==================
-let network = NETWORK.TESTNET; // <= NOTE: Update this when changing network
-const walletPrivateKey = process.env.SUI_TESTNET_BASE_64; // <= NOTE: Update this when changing network
-// ================================================================
-
-const registry = REGISTRY[network];
-const provider = new JsonRpcProvider(
-  new Connection({ fullnode: registry["RPC_URL"] })
-);
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+import { DefaultStore, SuiContract } from "contract_manager";
+import { PriceServiceConnection } from "@pythnetwork/price-service-client";
+
+const parser = yargs(hideBin(process.argv))
+  .usage(
+    "Usage: $0 --contract <contract_id> --feed-id <feed-id> --private-key <private-key>"
+  )
+  .options({
+    contract: {
+      type: "string",
+      demandOption: true,
+      desc: "Contract to update price feeds for (e.g sui_testnet_0xe8c2ddcd5b10e8ed98e53b12fcf8f0f6fd9315f810ae61fa4001858851f21c88)",
+    },
+    "feed-id": {
+      type: "array",
+      demandOption: true,
+      desc: "Price feed ids to create without the leading 0x (e.g f9c0172ba10dfa4d19088d94f5bf61d3b54d5bd7483a322a982e1373ee8ea31b). Can be provided multiple times for multiple feed updates",
```

Comment:
confirmed that this script works

wonder why it doesn't work when you include the leading `0x` when specifying price feed ids (in which case it assumes it's an array of numbers and won't convert it to the type `[]string` properly)

Reply:
It's because Javascript accepts `0x` prefixed strings as number literals